backend/qn-history-service/src/controllers/questionHistoryController.ts

```diff
@@ -80,74 +83,90 @@ export const updateQnHistory = async (
   }
 };
 
-export const deleteQnHistory = async (
-  req: Request,
-  res: Response
-): Promise<void> => {
-  try {
-    const { id } = req.params;
-
-    if (!id.match(MONGO_OBJ_ID_FORMAT)) {
-      res.status(400).json({ message: MONGO_OBJ_ID_MALFORMED_MESSAGE });
-      return;
-    }
-
-    const currQnHistory = await QnHistory.findById(id);
-    if (!currQnHistory) {
-      res.status(404).json({ message: QN_HIST_NOT_FOUND_MESSAGE });
-      return;
-    }
-
-    await QnHistory.findByIdAndDelete(id);
-    res.status(200).json({ message: QN_HIST_DELETED_MESSAGE });
-  } catch (error) {
-    res.status(500).json({ message: SERVER_ERROR_MESSAGE, error });
-  }
-};
-
 type QnHistListParams = {
   page: string;
   qnHistLimit: string;
   userId: string;
+  title: string; //qn title search keyword
+  status: string; //submission status
+  order: string; //entries sort order
```

Comment:
maybe we can define the types in another file cos right now its in the middle of the document